File: app/src/main/java/gardezi/io/amiibofinder/dagger/ViewModelSubComponent.java
package gardezi.io.amiibofinder.dagger;

import dagger.Subcomponent;
import gardezi.io.amiibofinder.viewmodel.AmiiboViewModel;

@Subcomponent
public interface ViewModelSubComponent {
    @Subcomponent.Builder
    interface Builder {
        ViewModelSubComponent build();
    }

    AmiiboViewModel amiiboViewModel();
}

File: app/src/main/java/gardezi/io/amiibofinder/dagger/ViewModelSubComponent.kt
package gardezi.io.amiibofinder.dagger

import dagger.Subcomponent
import gardezi.io.amiibofinder.viewmodel.AmiiboViewModel

@Subcomponent
interface ViewModelSubComponent {
    @Subcomponent.Builder
    interface Builder {
        fun build(): ViewModelSubComponent
    }

    fun amiiboViewModel(): AmiiboViewModel
}
